### PR TITLE
ci: use bazelisk to run the desired version of bazel

### PR DESCRIPTION
**Pull request by devjgm**
Related to: https://github.com/googleapis/google-cloud-cpp/issues/5878

Bazelisk (https://github.com/bazelbuild/bazelisk) seems to be the modern recommended way to get bazel (I think). So rather than installing a specific version of bazel ourselves, we're now using bazelisk to do that. Bazel can be run either via `bazel` or `bazelisk`. We use the `USE_BAZEL_VERSION` environment variable to tell bazelisk which version of bazel to use. We also cache the downloaded bazel installation.

I also changed the "Machine Info" section to no longer print the bazel and cmake versions. I did this because the bazel version was always wrong because the `USE_BAZEL_VERSION` env var wasn't set yet. So now the version of bazel/cmake is now printed by the relevant `builds/lib/` file, which seems reasonable.

We no longer need the `ci/install-bazel.sh` nor the `ci/etc/install-config.sh` scripts.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6734)

### CODE REVIEW

**Review by coryan — APPROVED** [review]
Would be nice to have these changes on Windows too, but feel free to do that in a separate PR.  Installing `npm` to just download a single `golang` binary seems like overkill, maybe?

File: ci/cloudbuild/dockerfiles/fedora-34.Dockerfile [hunk 1]
```diff
@@ -38,6 +38,8 @@ RUN dnf makecache && dnf install -y python3-devel
 RUN pip3 install --upgrade pip
 RUN pip3 install setuptools wheel
 
+RUN npm install -g @bazel/bazelisk
```
[the remark this answers is not in the document]
Comment (devjgm) [reply ci/cloudbuild/dockerfiles/fedora-34.Dockerfile:41]: Good idea. Done.

File: ci/cloudbuild/dockerfiles/ubuntu-bionic.Dockerfile [hunk 2]
```diff
@@ -65,5 +65,9 @@ ENV PATH=${CLOUD_SDK_LOCATION}/bin:${PATH}
 # The Cloud Pub/Sub emulator needs Java :shrug:
 RUN apt update && (apt install -y openjdk-11-jre || apt install -y openjdk-9-jre)
 
-# Install Bazel because some of the builds need it.
-RUN /var/tmp/ci/install-bazel.sh
+# Normally we'd install this with `npm install -g @bazel/bazelisk`, but
+# installing npm on Bionic fails due to a conflicting dependency version with
+# some other installed package.
+RUN curl -o /usr/bin/bazelisk -sSL "https://github.com/bazelbuild/bazelisk/releases/download/v1.9.0/bazelisk-linux-amd64" && \
```
[the remark this answers is not in the document]
Comment (devjgm) [reply ci/cloudbuild/dockerfiles/ubuntu-bionic.Dockerfile:71]: Good idea. Done.

[the remark this answers is not in the document]
Comment (devjgm) [reply ci/cloudbuild/dockerfiles/ubuntu-bionic.Dockerfile:68]: Now to figure out why `kokoro/macos/quickstart-bazel` keeps hanging at the end.

[the remark this answers is not in the document]
Comment (devjgm) [reply ci/cloudbuild/dockerfiles/ubuntu-bionic.Dockerfile:68]: I think I found the one missing `bazelisk shutdown` that I needed to add back.

**Review by coryan — APPROVED** [review]

**Pull request merged**